core/geotime/src/main/java/org/locationtech/geowave/core/geotime/store/query/filter/expression/spatial/PreparedFilterGeometry.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2013-2020 Contributors to the Eclipse Foundation
+ *
+ * <p> See the NOTICE file distributed with this work for additional information regarding copyright
+ * ownership. All rights reserved. This program and the accompanying materials are made available
+ * under the terms of the Apache License, Version 2.0 which accompanies this distribution and is
+ * available at http://www.apache.org/licenses/LICENSE-2.0.txt
+ */
+package org.locationtech.geowave.core.geotime.store.query.filter.expression.spatial;
+
+import org.locationtech.geowave.core.geotime.util.GeometryUtils;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.prep.PreparedGeometry;
+import org.locationtech.jts.geom.prep.PreparedGeometryFactory;
+
+public class PreparedFilterGeometry implements FilterGeometry {
+
+  private PreparedGeometry geometry;
+
+  public PreparedFilterGeometry() {}
+
+  public PreparedFilterGeometry(final PreparedGeometry geometry) {
+    this.geometry = geometry;
+  }
+
+  @Override
+  public Geometry getGeometry() {
+    return geometry.getGeometry();
+  }
+
+  @Override
+  public boolean intersects(FilterGeometry other) {
+    return geometry.intersects(other.getGeometry());
+  }
+
+  @Override
+  public boolean disjoint(FilterGeometry other) {
+    return geometry.disjoint(other.getGeometry());
+  }
+
+  @Override
+  public boolean crosses(FilterGeometry other) {
+    return geometry.crosses(other.getGeometry());
+  }
+
+  @Override
+  public boolean overlaps(FilterGeometry other) {
+    return geometry.overlaps(other.getGeometry());
+  }
+
+  @Override
+  public boolean touches(FilterGeometry other) {
+    return geometry.touches(other.getGeometry());
+  }
+
+  @Override
+  public boolean within(FilterGeometry other) {
+    return geometry.within(other.getGeometry());
+  }
+
+  @Override
+  public boolean contains(FilterGeometry other) {
+    return geometry.contains(other.getGeometry());
+  }
+
+  @Override
+  public boolean isEqualTo(FilterGeometry other) {
+    return geometry.getGeometry().equalsTopo(other.getGeometry());
+  }
+
+  @Override
+  public byte[] toBinary() {
+    return GeometryUtils.geometryToBinary(getGeometry(), null);
+  }
+
+  @Override
+  public void fromBinary(byte[] bytes) {
+    final Geometry unprepared = GeometryUtils.geometryFromBinary(bytes, null);
+    geometry = PreparedGeometryFactory.prepare(unprepared);
```

Comment:
reading the "interning" comment on SpatialQueryFilter I wonder whether this should use something similar like GeometryImage? (it uses it in fromBinary as well as as the constructor, but I don't know for sure how relevant it still is, seems like it could be worthy of a little investigation though based on the claims in performance gains).